net_utils.js

```diff
@@ -187,10 +180,59 @@ exports.is_ip_in_str = function(ip, str) {
     return false;
 };
 
-exports.is_rfc1918 = function (ip) {
-    return (net.isIPv4(ip) && re_private_ipv4.test(ip));
+exports.is_private_ipv4 = function (ip) {
+
+    // RFC 1918, reserved as "private" IP space
+    if (/^10\./.test(ip)) return true;          // 10/8
+    if (/^192\.168\./.test(ip)) return true;    // 192.168/16
+    // 172.16/12  ->   172.16/16 .. 172.31/16
+    if (/^172\.(1[6-9]|2[0-9]|3[01])\./.test(ip)) return true;
+
+    return false;
+};
+
+exports.is_local_ipv4 = function (ip) {
+    // 127/8 (loopback) and 0/8 (this network)  # RFC 1122
+    if(/^(127|0)\./.test(ip)) return true;
```

Comment:
This is wrong IMO.   0.0.0.0/8 shouldn't be included here.

RFC 1122 states:
`````
Specified host on this network. It MUST NOT be sent, except as a source address as part of an initialization procedure by which the host learns its full IP address.
`````

So you'll never see hosts with 0.0.0.0 addresses connecting to Haraka and if you do; they certainly shouldn't be treated specially, so I don't think it should be present.